docs/painless/painless-contexts/painless-watcher-context-variables.asciidoc

```diff
@@ -0,0 +1,158 @@
+The following variables are available in all watcher contexts.
+
+*Variables*
+
+`params` (`Map`, read-only)::
+        User-defined parameters passed in as part of the query.
+
+`ctx['watch_id']` (`String`, read-only)::
+        The id of the watch.
+
+`ctx['id']` (`String`, read-only)::
+        The server generated unique identifer for the run watch.
+
+`ctx['metadata']` (`Map`, read-only)::
+        Any metadata associated with the watch.
+
+`ctx['execution_time']` (`ZonedDateTime`, read-only)::
+        The start time for the watch.
+
+`ctx['trigger']['scheduled_time']` (`ZonedDateTime`, read-only)::
+        The scheduled trigger time for the watch.
+
+`ctx['trigger']['triggered_time']` (`ZonedDateTime`, read-only)::
+        The actual trigger time for the watch.
+
+`ctx['payload']` (`Map`, read-only)::
+        The accessible watch data based upon the
+        {xpack-ref}/input.html[watch input].
+
+*API*
+
+
+The standard <<painless-api-reference, Painless API>> is available.
+
+To run this example, first follow the steps in
+<<painless-context-examples, context examples>>.
+
+==== Metadata in a watch
+
+Metadata can be added to the top level of the watch definition. This
+is user defined and is typically used to consolidate duplicate values
+in a watch.
+
+[source,Painless]
+----
+POST _watcher/watch/_execute
+{
+  "watch" : {
+    "metadata" : { "high_threshold": 50000 },
+    "trigger" : { "schedule" : { "interval" : "24h" } },
+    "condition" : {
+      "script" :
+      """
+        return ctx.metadata.high_threshold == 50000     <1>
```

Comment:
this example does not make sense. Maybe have a search input, that checks against `ctx.payload.hits.total`?

Reply:
This and the time values examples seem overly-heavyweight to simply show how to access these values. I'd incorporate the info/example snippet into the context variable descriptions and maybe mention in the intro that condition scripts can access watch metadata and timestamps.